Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -2058,6 +2059,391 @@ - (void)setMinimumViewportInset:(CocoaEdgeInsets)minimumViewportInset maximumVie
+        return WTTextSuggestionStateReviewing;
+    case WebCore::UnifiedTextReplacement::Replacement::State::Reverted:
+        return WTTextSuggestionStateRejected;
+    case WebCore::UnifiedTextReplacement::Replacement::State::Invalid:
+        return WTTextSuggestionStateInvalid;
+    }
+}
+
+static WebCore::UnifiedTextReplacement::Replacement::State convert(WTTextSuggestionState state)
+{
+    switch (state) {
+    case WTTextSuggestionStatePending:
+        return WebCore::UnifiedTextReplacement::Replacement::State::Pending;
+    case WTTextSuggestionStateReviewing:
+        return WebCore::UnifiedTextReplacement::Replacement::State::Active;
+    case WTTextSuggestionStateRejected:
+        return WebCore::UnifiedTextReplacement::Replacement::State::Reverted;
+    case WTTextSuggestionStateInvalid:
+        return WebCore::UnifiedTextReplacement::Replacement::State::Invalid;
+
+    // FIXME: Remove this default case once the WTTextSuggestionStateAccepted case is no longer in the build.
+    default:
+        ASSERT_NOT_REACHED();
+        return WebCore::UnifiedTextReplacement::Replacement::State::Invalid;
+    }
+}
+
+static WebCore::UnifiedTextReplacement::EditAction convert(WTAction action)
+{
+    switch (action) {
+    case WTActionShowOriginal:
+        return WebCore::UnifiedTextReplacement::EditAction::Undo;
+    case WTActionShowRewritten:
+        return WebCore::UnifiedTextReplacement::EditAction::Redo;
+    case WTActionCompositionRestart:
+        return WebCore::UnifiedTextReplacement::EditAction::UndoAll;
+    }
+}
+
+static WebCore::UnifiedTextReplacement::Session::ReplacementType convert(WTSessionType type)
+{
+    switch (type) {
+    case WTSessionTypeProofreading:
+        return WebCore::UnifiedTextReplacement::Session::ReplacementType::PlainText;
+    case WTSessionTypeComposition:
+        return WebCore::UnifiedTextReplacement::Session::ReplacementType::RichText;
+    }
+}
+
+static WebCore::UnifiedTextReplacement::Session::CorrectionType convert(WTCompositionSessionType type)
+{
+    switch (type) {
+    case WTCompositionSessionTypeNone:
+        return WebCore::UnifiedTextReplacement::Session::CorrectionType::None;
+
+    // FIXME: Map these to specific `CorrectionType` types post-upstreaming.
+    case WTCompositionSessionTypeMagic:
+    case WTCompositionSessionTypeConcise:
+    case WTCompositionSessionTypeFriendly:
+    case WTCompositionSessionTypeProfessional:
+    case WTCompositionSessionTypeOpenEnded:
+    case WTCompositionSessionTypeSummary:
+    case WTCompositionSessionTypeKeyPoints:
+    case WTCompositionSessionTypeList:
+    case WTCompositionSessionTypeTable:
+    case WTCompositionSessionTypeCompose:
+        return WebCore::UnifiedTextReplacement::Session::CorrectionType::Grammar;
+
+    case WTCompositionSessionTypeSmartReply:
+        return WebCore::UnifiedTextReplacement::Session::CorrectionType::Spelling;
+
+    default:
```

Comment:
Can we remove this?

Reply:
not yet (there are some cases where things won't build if we remove it)